src/modules/api/user/user.service.ts

```diff
@@ -2,8 +2,14 @@ import { Injectable } from '@nestjs/common';
 import { InjectModel } from '@nestjs/mongoose';
 import { User } from 'src/schemas/user.schema';
 import { Model } from 'mongoose';
+import { UserDTO } from './user.dto';
 
 @Injectable()
 export class UserService {
   constructor(@InjectModel(User.name) private userModel: Model<User>) {}
+
+  async getUserById(id: string): Promise<UserDTO> {
+    const user = await this.userModel.findById(id);
```

Comment:
You'll receive IUser ?

Reply:
No, User object

Reply:
UserDTO is what u'll be returning as response.